openshift-with-appstudio-test/e2e/periodic_test.go

```diff
@@ -144,25 +167,53 @@ func TestServiceRegistry(t *testing.T) {
 				if !ok {
 					continue
 				}
-				s, k := state.Load(db.Name)
+				stateMutex.Lock()
+				//so there is the possibility of a race here
+				//if a DB has quick state changes we can get two concurrent goroutines for the same DependencyBuild
+				//if they are processed out of order then we could essentually miss the build going into its final state
+				//to get around this we use a Mutex, and query the current resource version. If this matches then we know
+				//the version we are using is up to date, and update the state map accordingly
+				//note that we don't need the mutex for the add/delete code below, as even if these do race the end
+				//net result will be correct as the state changes themselves will be processed in the correct order
+				s, k := stateMap[db.Name]
+				current, err := jvmClient.JvmbuildserviceV1alpha1().DependencyBuilds(ta.ns).Get(ctx, db.Name, metav1.GetOptions{})
+				if err != nil {
+					ta.Logf("Failed to get latest DependencyBuild version" + err.Error())
+					stateMutex.Unlock()
+					break
+				}
+				if current.ResourceVersion != db.ResourceVersion {
+					ta.Logf(fmt.Sprintf("Skipping watch event as DependencyBuild %s was not the most recent version", db.Name))
```

Comment:
Looking at how many times this message is generated I think this is clearly not just theoretical.